timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosTimestampBoundStore.java

```diff
@@ -246,15 +253,9 @@ public synchronized void storeUpperLimit(long limit) throws MultipleRunningTimes
                     // This is dangerous; proposing at the next sequence number is unsafe, as timestamp services
                     // generally assume they have the ALLOCATION_BUFFER_SIZE timestamps up to this.
                     // TODO (jkong): Devise a method that better preserves availability of the cluster.
-                    log.warn(
-                            "It appears we updated the timestamp limit to {}, which was less than our target {}."
-                                    + " This suggests we have another timestamp service running; possibly because we"
-                                    + " lost and regained leadership. For safety, we are now stopping this service.",
-                            SafeArg.of("newLimit", newLimit),
-                            SafeArg.of("target", limit));
-                    throw new NotCurrentLeaderException(String.format(
-                            "We updated the timestamp limit to %s, which was less than our target %s.",
-                            newLimit, limit));
+                    List<Arg<?>> args = List.of(SafeArg.of("newLimit", newLimit), SafeArg.of("target", limit));
+                    log.warn(TOO_SMALL_LIMIT_INCREASE_MESSAGE, args);
+                    throw new MaybeNotCurrentLeaderException(TOO_SMALL_LIMIT_INCREASE_MESSAGE, args);
```

Comment:
Not tested for, as this is rather difficult to simulate these conditions, which is probably why this was not tested prior.